plugins/org.jboss.tools.openshift.express.core/src/org/jboss/tools/openshift/express/internal/core/connection/ExpressConnectionPersistency.java

```diff
@@ -25,7 +29,10 @@
 
 	@Override
 	protected String[] loadPersisted() {
-		return ExpressCorePreferences.INSTANCE.loadConnections();
+		List<String> connections = new ArrayList<String>();
+		connections.addAll(Arrays.asList(ExpressCorePreferences.INSTANCE.loadConnections()));
+		connections.addAll(Arrays.asList(ExpressCorePreferences.INSTANCE.loadLegacyConnections()));
+		return connections.toArray(new String[connections.size()]);
```

Comment:
that's a convoluted way to concatenate arrays. Rather use commons-lang ArrayUtils.addAll(first, second);